test/test_autograd.py

```diff
@@ -1752,6 +1752,23 @@ def test_backward_twice_retained_graph_without_saved_values(self):
         c.backward(torch.tensor([1, 1, 1], dtype=torch.double), retain_graph=True)
         c.backward(torch.tensor([1, 1, 1], dtype=torch.double))
 
+    def test_backward_create_graph_warns(self):
+        prev = torch.is_warn_always_enabled()
+        torch.set_warn_always(True)
+
+        b = torch.randn(3, requires_grad=True, dtype=torch.double)
+        c = b * b
+        with warnings.catch_warnings(record=True) as ws:
+            c.backward(torch.ones_like(c), retain_graph=True ,create_graph=True)
+        self.assertTrue(any('Using backward() with create_graph=True' in str(w.message) for w in ws))
+
+        # Should not warn for grad
+        with warnings.catch_warnings(record=True) as ws:
+            torch.autograd.grad(c, b, torch.ones_like(c), create_graph=True)
+        self.assertFalse(any('Using backward() with create_graph=True' in str(w.message) for w in ws))
+
+        torch.set_warn_always(prev)
```

Comment:
This would be best inside a `try/finally` block to make sure it is always running.

Reply:
Good point! Not sure there's an analog for the cpp case though. We'd need an RAII version of `c10::Warning::set_warnAlways`.

Reply:
That sounds like an ok thing to add indeed :)

Reply:
Added. 
Let me know though if you think it should be split into its own PR

Reply:
That's ok now that it is here.